### PR TITLE
docs: rhai example for header->client propagation

### PR DESCRIPTION
**Pull request by lennyburdette**
Shows how to send set-cookie headers from subgraphs to clients with rhai scripting.

Related to #1284

- Fix #issue_number

*description here*

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [ ] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]. It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]. Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]. Tick whichever testing boxes are applicable. If you are adding Manual Tests:
    - please document the manual testing (extensively) in the Exceptions.
    - please raise a separate issue to automate the test and label it (or ask for it to be labeled) as `manual test`

### CODE REVIEW

**Review by bnjjj — APPROVED** [review]

File: docs/source/configuration/header-propagation.mdx [hunk 1]
```diff
@@ -206,23 +206,74 @@ headers:
             value: "accounts"
 ```
 
+## Response header propagation
+
+It is not currently possible to propagate response headers from subgraphs to clients using YAML configuration alone. It can however be achieved using [Rhai scripting]("../customizations/rhai").
+
+This approach relies the fact that all requests have context that can be used to store data for the duration of the request:
```
Comment (ebylund) [inline docs/source/configuration/header-propagation.mdx:213]:
```suggestion
This approach relies on the fact that all requests have context that can be used to store data for the duration of the request:
```

File: docs/source/configuration/header-propagation.mdx [hunk 2]
```diff
@@ -206,23 +206,74 @@ headers:
...
+If you would like to see a YAML based solution then [please leave us a comment on our issue tracker](https://github.com/apollographql/router/issues/1284).
 
 ## Propagation between subgraphs
 
 It is not currently possible to propagate headers between subgraphs using yaml config alone. It can however be achieved using [Rhai scripting]("../customizations/rhai").
 
-The approach relied on the feature that all requests have context that can be used to store data for the duration of the request: 
-1. On supbgraph response, copy header values into context.
-2. On supbgraph request, copy header values from context into the subgraph request.
+This approach relies the fact that all requests have context that can be used to store data for the duration of the request:
```
Comment (ebylund) [inline docs/source/configuration/header-propagation.mdx:263]:
```suggestion
This approach relies on the fact that all requests have context that can be used to store data for the duration of the request:
```

**Review by garypen — APPROVED** [review]
Approving, but I did spot one edit worth applying.

File: docs/source/configuration/header-propagation.mdx [hunk 3]
```diff
@@ -206,23 +206,74 @@ headers:
...
 ```
 
+## Response header propagation
+
+It is not currently possible to propagate response headers from subgraphs to clients using YAML configuration alone. It can however be achieved using [Rhai scripting]("../customizations/rhai").
+
+This approach relies on the fact that all requests have context that can be used to store data for the duration of the request:
+
+1. For each subgraph response, copy header values into context.
+2. For each the supergraph response, copy header values from the context onto the response.
```
[the remark this answers is not in the document]
Comment (lennyburdette) [reply docs/source/configuration/header-propagation.mdx]: good catch, thanks! are you able to squash-and-merge this or should i rebase and merge the commits manually?

**Pull request merged**